### PR TITLE
feature: added --- skip_check_config_version to skip check config_version after HUP reload.

### PR DESCRIPTION
**Pull request by doujiang24**
Yeah, a little hacky here, because we need this to test the following cases:
https://github.com/openresty/lua-nginx-module/pull/1002

### CODE REVIEW

**Comment by agentzh** [comment]
@doujiang24 Will you elaborate?

**Comment by agentzh** [comment]
@doujiang24 Okay, after seeing your other PR, openresty/lua-nginx-module#1002, I see what you mean here.

I think `--- reload_fails` is a better name here. We should use names here for test semantics instead of for mechanisms :)

**Comment by doujiang24** [comment]
@agentzh Fixed :)

**Comment by agentzh** [comment]
@doujiang24 Mind to add some documentation to `Test::Nginx::Socket`'s POD?

**Comment by doujiang24** [comment]
@agentzh Added some document & rebased :)

File: lib/Test/Nginx/Socket.pm [hunk 1]
```diff
@@ -3316,6 +3316,10 @@ Below is an example from ngx_headers_more module's test suite:
 
 Do not attempt to parse the response or run the response related subtests.
 
+=head2 reload_fails
+
+Allow HUP reload fails, that means the server still use the previous config.
```
Comment (agentzh) [inline lib/Test/Nginx/Socket.pm]: English issues: ", that means" => ", which means"

"the server still use" => "the server still uses"

**Comment by agentzh** [comment]
@doujiang24 Merged with the following patch:

```diff
diff --git a/lib/Test/Nginx/Socket.pm b/lib/Test/Nginx/Socket.pm
index bee7673..9eeed94 100644
--- a/lib/Test/Nginx/Socket.pm
+++ b/lib/Test/Nginx/Socket.pm
@@ -3318,7 +3318,9 @@ Do not attempt to parse the response or run the response related subtests.

 =head2 reload_fails

-Allow HUP reload fails, that means the server still use the previous config.
+Allows the NGINX HUP reload fails, which means that the server will still use the previous test block's nginx configuration.
+
+This only makes sense in the HUP reload testing mode.

 =head2 user_files

```

Also updated your commit log message to the following:

```
    feature: added the --- reload_fails new section to skip checking config_version after HUP reload in the HUp reload testing mode.

    We need this to test failures in init_by_lua*.
```

**Pull request closed without merging**